src/EditorFeatures/TestUtilities2/Intellisense/ModernCompletionTestState.vb

```diff
@@ -0,0 +1,349 @@
+        End Sub
+
+        Public Overrides Sub SendSurroundWithCommand()
+            ' The legacy handler implements VSCommanding.IChainedCommandHandler(Of SurroundWithCommandArgs)
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of SurroundWithCommandArgs))
+            MyBase.SendSurroundWithCommand(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
+        End Sub
+
+        Public Overrides Sub SendSave()
+            ' The legacy handler implements VSCommanding.IChainedCommandHandler(Of SaveCommandArgs)
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of SaveCommandArgs))
+            MyBase.SendSave(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
+        End Sub
+
+        Public Overrides Sub SendSelectAll()
+            ' The legacy handler implements VSCommanding.IChainedCommandHandler(Of SelectAllCommandArgs)
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of SelectAllCommandArgs))
+            MyBase.SendSelectAll(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
+        End Sub
+
+        Public Overrides Sub SendDeleteWordToLeft()
+            Dim compHandler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of WordDeleteToStartCommandArgs))
+            MyBase.SendWordDeleteToStart(Sub(a, n, c) compHandler.ExecuteCommand(a, n, c), AddressOf MyBase.SendDeleteWordToLeft)
+        End Sub
+
+        Protected Overrides Function GetHandler(Of T As VSCommanding.ICommandHandler)() As T
+            Return DirectCast(EditorCompletionCommandHandler, T)
+        End Function
+
+#End Region
+
+#Region "Completion Operations"
+
+        Public Overrides Sub SendCommitUniqueCompletionListItem()
+            ' The legacy handler implements VSCommanding.IChainedCommandHandler(Of CommitUniqueCompletionListItemCommandArgs)
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of CommitUniqueCompletionListItemCommandArgs))
+            MyBase.SendCommitUniqueCompletionListItem(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
+        End Sub
+
+        Public Overrides Async Function AssertNoCompletionSession(Optional block As Boolean = True) As Task
+            If block Then
+                Await WaitForAsynchronousOperationsAsync()
+            End If
+            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(TextView)
+            If session Is Nothing Then
+                Return
+            End If
+
+            If session.IsDismissed Then
+                Return
+            End If
+
+            ' If completionItems cannot be calculated in 5 seconds, no session exists.
+            If Not block Then
+                Dim task1 = New Task(Sub()
+                                         Task.Delay(5000)
```

Comment:
This is broken now, since the task returned from Task.Delay is being thrown away. You can just remove the lambda entirely since the task being returned from Task.Delay is precisely what you want here.